src/unix.rs

```diff
@@ -42,6 +45,17 @@ fn dup_fd(fd: RawFd) -> io::Result<Stdio> {
     dup_result.map(File::into_stdio)
 }
 
+fn nix_err_to_io_err(err: nix::Error) -> io::Error {
+    match err {
+        Sys(err_no) => {
+            io::Error::from(err_no)
+        }
+        _ => {
+            io::Error::new(ErrorKind::InvalidData, err)
+        }
+    }
+}
```

Comment:
The conversion was removed in nix as part of https://github.com/nix-rust/nix/pull/614
And discussed here https://github.com/nix-rust/nix/issues/613 (see the end of the thread 22Jul17) - so it seems to be intentional. 

I'm not involved with the nix project, but there is talk about improving the structure of error handling in nix in the same threads mentioned above, so hopefully there will be something nicer in future.